parser.y

```diff
@@ -7825,109 +7827,111 @@ SelectLockOpt:
-UnionStmt:
-	UnionClauseList "UNION" UnionOpt SelectStmtBasic OrderByOptional SelectStmtLimit SelectLockOpt
-	{
-		st := $4.(*ast.SelectStmt)
-		union := $1.(*ast.UnionStmt)
-		st.IsAfterUnionDistinct = $3.(bool)
-		lastSelect := union.SelectList.Selects[len(union.SelectList.Selects)-1]
-		endOffset := parser.endOffset(&yyS[yypt-5])
+// See https://mariadb.com/kb/en/intersect/
+// See https://mariadb.com/kb/en/except/
+SetOprStmt:
+	SetOprClauseList SetOpr SelectStmtBasic OrderByOptional SelectStmtLimit SelectLockOpt
+	{
+		st := $3.(*ast.SelectStmt)
+		setOpr := $1.(*ast.SetOprStmt)
+		st.AfterSetOperator = $2.(*ast.SetOprType)
+		lastSelect := setOpr.SelectList.Selects[len(setOpr.SelectList.Selects)-1]
+		endOffset := parser.endOffset(&yyS[yypt-4])
 		parser.setLastSelectFieldText(lastSelect, endOffset)
-		union.SelectList.Selects = append(union.SelectList.Selects, st)
-		if $5 != nil {
-			union.OrderBy = $5.(*ast.OrderByClause)
+		setOpr.SelectList.Selects = append(setOpr.SelectList.Selects, st)
+		if $4 != nil {
+			setOpr.OrderBy = $4.(*ast.OrderByClause)
 		}
-		if $6 != nil {
-			union.Limit = $6.(*ast.Limit)
+		if $5 != nil {
+			setOpr.Limit = $5.(*ast.Limit)
 		}
-		if $5 == nil && $6 == nil {
-			st.LockTp = $7.(ast.SelectLockType)
+		if $4 == nil && $5 == nil {
+			st.LockTp = $6.(ast.SelectLockType)
 		}
-		$$ = union
+		$$ = setOpr
 	}
-|	UnionClauseList "UNION" UnionOpt SelectStmtFromDualTable OrderByOptional SelectStmtLimit SelectLockOpt
+|	SetOprClauseList SetOpr SelectStmtFromDualTable OrderByOptional SelectStmtLimit SelectLockOpt
 	{
-		st := $4.(*ast.SelectStmt)
-		union := $1.(*ast.UnionStmt)
-		st.IsAfterUnionDistinct = $3.(bool)
-		lastSelect := union.SelectList.Selects[len(union.SelectList.Selects)-1]
-		endOffset := parser.endOffset(&yyS[yypt-5])
+		st := $3.(*ast.SelectStmt)
+		setOpr := $1.(*ast.SetOprStmt)
+		st.AfterSetOperator = $2.(*ast.SetOprType)
+		lastSelect := setOpr.SelectList.Selects[len(setOpr.SelectList.Selects)-1]
+		endOffset := parser.endOffset(&yyS[yypt-4])
 		parser.setLastSelectFieldText(lastSelect, endOffset)
-		union.SelectList.Selects = append(union.SelectList.Selects, st)
-		if $5 != nil {
-			union.OrderBy = $5.(*ast.OrderByClause)
+		setOpr.SelectList.Selects = append(setOpr.SelectList.Selects, st)
+		if $4 != nil {
+			setOpr.OrderBy = $4.(*ast.OrderByClause)
 		}
-		if $6 != nil {
-			union.Limit = $6.(*ast.Limit)
+		if $5 != nil {
+			setOpr.Limit = $5.(*ast.Limit)
 		}
-		if $5 == nil && $6 == nil {
-			st.LockTp = $7.(ast.SelectLockType)
+		if $4 == nil && $5 == nil {
+			st.LockTp = $6.(ast.SelectLockType)
 		}
-		$$ = union
+		$$ = setOpr
 	}
-|	UnionClauseList "UNION" UnionOpt SelectStmtFromTable OrderByOptional SelectStmtLimit SelectLockOpt
+|	SetOprClauseList SetOpr SelectStmtFromTable OrderByOptional SelectStmtLimit SelectLockOpt
 	{
-		st := $4.(*ast.SelectStmt)
-		union := $1.(*ast.UnionStmt)
-		st.IsAfterUnionDistinct = $3.(bool)
```

Comment:
can we ignore this?

Reply:
I used the `afterSetOprType` instead of it.